### ISSUE TITLE
Does not support es6 destructive assignment in method arguments

### ISSUE THREAD

**Issue by koorchik** [jshint/jshint#1796]
with "esnext: true" on version 2.5.3
```
class Test {
    update({ groups = [] }) {
    }
}
```
produces error
```
Expected ',' and instead saw ']'. @ line 15 char 24
        update({ groups = [] }) {
```

**Comment by ms88privat** [comment]
+1, thank you

**Comment by nicolo-ribaudo** [comment]
Fixed by #2391

**Comment by lukeapage** [comment]
That PR has been merged and this will be fixed in the next release after 2.8.0
